core/citrus-api/src/main/java/com/consol/citrus/validation/AbstractMessageValidator.java

```diff
@@ -39,11 +39,14 @@
     @Override
     public final void validateMessage(Message receivedMessage, Message controlMessage, TestContext context,
             List<ValidationContext> validationContexts) throws ValidationException {
-        T validationContext = findValidationContext(validationContexts);
-
-        // check if we were able to find a proper validation context
-        if (validationContext != null) {
-            validateMessage(receivedMessage, controlMessage, context, validationContext);
+        
+        // validate all matching contexts
+        for (ValidationContext validationContext : validationContexts) {
+            if (getRequiredValidationContextType().isInstance(validationContext)) {
+                if (validationContext != null) {
+                    validateMessage(receivedMessage, controlMessage, context, validationContext);
```

Comment:
this does not match the `validateMessage(Message, Message, TestContext, T)` methods pattern and [does not compile](https://github.com/citrusframework/citrus/actions/runs/3272665059/jobs/5426109978). you must cast the `validationContext` after you already type-checked it:
```suggestion
                    validateMessage(receivedMessage, controlMessage, context, getRequiredValidationContextType().cast(validationContext));
```